### DISCUSSION TITLE
Dark mode toggle not working

### DISCUSSION THREAD

**Discussion by m40ma0** [tailwindlabs/tailwindcss#9600] [Help]
I have tried various techniques, but I cant get my dark mode toggle to work.

Link to REPL is here: https://svelte.dev/repl/3281de8654ec42bfbc66345a91af209b?version=3.18.1

App.html code:
`<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="utf-8" />
		<link rel="icon" href="/ux.png" />
		<meta name="viewport" content="width=device-width, initial-scale=1" />
		%sveltekit.head%
		
	</head>
	<body class=" dark:bg-[#03045E]" style="background-color: #FBF8CC">
		<div>%sveltekit.body%</div>
	</body>
</html>`

tailwind.config.cjs code:
`/** @type {import('tailwindcss').Config} */
module.exports = {
  darkMode: 'class',
  mode: 'jit',
  content: ['./src/**/*.svelte'],
  theme: {
    extend: {},
  },
  plugins: [require("daisyui")],
  daisyui: {
    themes: ['light', 'dark']
  },
}`

**Comment by wongjn** [comment]
Your `toggleMode()` function in `nav.svelte` never gets called since it is an event listener callback on the `click` event of the `dark-toggle.svelte` component, but this component never emits the `click` event. You need to emit the event from `dark-toggle.svelte` with [`createEventDispatcher()`](https://svelte.dev/docs#run-time-svelte-createeventdispatcher) or by [forwarding the event](https://svelte.dev/docs#template-syntax-component-directives-on-eventname). Further, the `toggleMode()` function seems to have some code that should be run `onMount` or similar for the initial theme choice.

**Reply by m40ma0** [reply]
Hi, what is createEventDispatcher and what exactly does onMount do?

**Reply by wongjn** [reply]
[From the documentation for `createEventDispatcher()`](https://svelte.dev/docs#run-time-svelte-createeventdispatcher):
> Creates an event dispatcher that can be used to dispatch [component events](https://svelte.dev/docs#template-syntax-component-directives-on-eventname).

[From the documentation for `onMount()`](https://svelte.dev/docs#run-time-svelte-onmount):
> The `onMount` function schedules a callback to run as soon as the component has been mounted to the DOM.
